### ISSUE TITLE
Design and add a useful CLI

### ISSUE THREAD

**Issue by lebrice** [mila-iqia/ResearchTemplate#28]
We'd need to think about the kinds of commands that people will be running, and what kind of things we can setup for them.

Ideas:
- `project sync` (sync the codebase across all the clusters so it can be used for jobs)
- `project train (...)`: training run, with a validation set when applicable. No testing.
- `project eval (...)`: training (without a validation set), (or load from a checkpoint) then evaluate on the test set.
    - I'd ideally like to encourage not using the test set for validation whenever possible.
    - Other "good ML practices" should also be encouraged or even enforced, if that's desirable.
- `project profile (...)` same as `train`, but shorter, with more profiling tools setup?

Idk, just throwing ideas out there.
